File: api-tools/src/test/scala/com/lightbend/lagom/api/tools/tests/scaladsl/NoAclService.scala
/*
 * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
 */
package com.lightbend.lagom.api.tools.tests.scaladsl

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.transport.Method
import com.lightbend.lagom.scaladsl.api.{ Descriptor, Service, ServiceCall }
import com.lightbend.lagom.scaladsl.api.Service._

import scala.concurrent.Future

trait NoAclService extends Service {
  def getMock(id: String): ServiceCall[NotUsed, NotUsed]

  def descriptor: Descriptor =
    named("/noaclservice").withCalls(
      restCall(Method.GET, "/mocks/:id", getMock _)
    )
}

class NoAclServiceImpl extends NoAclService {
  def getMock(id: String) = ServiceCall { _ => Future.successful(NotUsed) }
}

File: api-tools/src/test/java/com/lightbend/lagom/api/tools/tests/javadsl/NoAclService.java
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package com.lightbend.lagom.api.tools.tests.javadsl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.transport.Method;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.restCall;

public interface NoAclService extends Service {

    ServiceCall<NotUsed, NotUsed> getMock(String id);

    default Descriptor descriptor() {
        return named("/noaclservice").withCalls(
            restCall(Method.GET,  "/mocks/:id", this::getMock)
        );
    }
}
